crates/services/p2p/src/service.rs

```diff
@@ -880,7 +887,16 @@ where
                             tracing::warn!("No peers found for block at height {:?}", height);
                         }
                     }
-                    Some(TaskRequest::GetTransactions { block_height_range, from_peer, channel }) => {
+                    Some(TaskRequest::GetTransactions {block_height_range, channel }) => {
+                        let channel = ResponseSender::Transactions(channel);
+                        let request_msg = RequestMessage::Transactions(block_height_range.clone());
+                        let height = BlockHeight::from(block_height_range.end.saturating_sub(1));
+                        let peer = self.p2p_service.get_peer_id_with_height(&height);
+                        if self.p2p_service.send_request_msg(peer, request_msg, channel).is_err() {
```

Comment:
We should return error to the `channel` if `peer` is `None`, to allow proper handling of it by the caller.

Reply:
Before doing it I want to ask if we want the same in `GetSealedHeaders` (just above in the code) ? It has the same behavior and doesn't return the error for now.

Reply:
Done. It complexify a bit the type in the channel because we have a two-stage error type but with unification of return type etc that we want to do in p2p maybe we could improve this in the future.